@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -34,91 +31,86 @@ const neutralColor = new THREE.Color(1, 1, 1);
 const tmpIconColor = new THREE.Color();
 
 /**
- * The `PoiRenderBufferBatch` contains the geometry and the material for all POIs that share the
- * same icon image ({@link @here/harp-datasource-protocol#ImageTexture}).
- *
- * @remarks
- * If the image is the same, all the objects in this batch can
- * share the same material, which makes them renderable in the same draw call, whatever the number
- * of actual objects (WebGL limits apply!).
- *
- * There is a `PoiRenderBufferBatch` for every icon in a texture atlas, since the size of the icon
- * in the atlas as well as the texture coordinates are specified in the `PoiRenderBufferBatch`.
+ * @internal
+ * Buffer for POIs sharing same material and render order, renderable in a single draw call
+ * (WebGL limits apply, see {@link BoxBuffer}).
  */
-class PoiRenderBufferBatch {
-    // Enable trilinear filtering to reduce flickering due to distance scaling
-    static trilinear: boolean = true;
-
-    boxBuffer: BoxBuffer | undefined;
-
-    private m_material?: THREE.Material | THREE.Material[];
+export class PoiBuffer {
```

Comment:
This is the class where we do reference counting and propagate disposal of any resources when ref count reaches 0.